### ISSUE TITLE
Update file names in documentation and code 

### ISSUE THREAD

**Issue by 3mma-mack** [riboviz/riboviz#411]
Having reviewed file names and discussed some alternate names in [issue 395](https://github.com/riboviz/riboviz/issues/395#issuecomment-896963460) we should start thinking about updating documentation and code to use these names on a new branch. 

* [x] update output file names in `docs/user/prep-riboviz-operation.md` and link that to `docs/user/riboviz_outputs.md`
* [x] update `prep_riboviz.nf` and/or `generate_stats.figs.R` and the static html to account for new names. Be prepared to grep the repository.
* [x] update system tests to account for new file names.

I have created a txt file, containing the names that we have decided on so far.

[file_name_changes_txt.txt](https://github.com/riboviz/riboviz/files/6974358/file_name_changes_txt.txt)

**Comment by 3mma-mack** [comment]
I have created the branch update-file-names-411 from develop to hold any changes to file names in the documentation and code.

**Comment by mikej888** [comment]
Other necessary documentation updates include:

* `docs/user/run-vignette.md`
* `docs/user/prep-riboviz-config.md`
* `docs/user/run-dedup-demultiplex-examples.md`
* `docs/images/*.dot` (once updated `*.svg` image files need updated also)
* `docs/developer/debugging.md`
* `docs/developer/testing.md` (sample output from integration test run, can be updated once code is updated)
* `docs/user/prep-riboviz-run-nextflow.md`

File name constants and patterns are listed in the following Python files.

* `riboviz/workflow_files.py`
* `riboviz/workflow_r.py`

Changing these should take care of the tests.

The Nextflow workflow, `prep_riboviz.nf`, as each task has to declare the names of the files it outputs. See the `generateStatsFigs`, `renameTpms`, `collateTpms`, `countReads`, `staticHTML` processes.

**Comment by mikej888** [comment]
Another cosmetic, but very useful, change is to update Python and R constants, variables and Nextflow channel names and variables that represent these files to be consistent with the new file names.

For example `riboviz/workflow_r.py` has
```
SEQUENCE_FEATURES_TSV = "sequence_features.tsv"
```
and `prep_riboviz.nf` `generateStatsFigs` declares output:
```
        tuple val(sample_id), file("sequence_features.tsv") \
            optional (! is_features_file) into sequence_features_tsv
```
As `sequence_features.tsv` is being renamed to ORF_TPMs_vs_features.tsv`, `workflow_r.py` would be changed to:
```
SEQUENCE_FEATURES_TSV = "ORF_TPMs_vs_features.tsv"
```
and  `generateStatsFigs`'s output to:
```
        tuple val(sample_id), file("ORF_TPMs_vs_features.tsv") \
            optional (! is_features_file) into sequence_features_tsv
```
But ideally the Python constant and Nextflow output channel name would also be changed to preserve consistency e.g.
```
ORF_TPMS_VS_FEATURES_TSV = "ORF_TPMs_vs_features.tsv"
```
and
```
        tuple val(sample_id), file("ORF_TPMs_vs_features.tsv") \
            optional (! is_features_file) into orf_tpms_vs_features_tsv
```

**Comment by mikej888** [comment]
As a suggestion, rather than updating all the doc, then all the code, take a file-by-file approach e.g. make all doc and code updates for `sequence_features.tsv`, run all tests, commit, repeat with next file.

**Comment by mikej888** [comment]
| Old name | New name | Developer | Done |
| -------- | -------- | --------- | ---- |
| `TPMs_collated.tsv` | `TPMs_all_CDS_all_samples.tsv` | - | - |
| `read_counts.tsv` | `read_counts_per_file.tsv` | - | - |
| `3nt_periodicity.pdf` | `metagene_start_stop_read_counts.pdf` | - | - |
| `3nt_periodicity.tsv` | `metagene_start_stop_read_counts.tsv` | - | - |
| `pos_sp_rpf_norm_reads.pdf` | `metagene_normalized_profile_start_stop.pdf` | - | - |
| `pos_sp_rpf_norm_reads.tsv` | `metagene_normalized_profile_start_stop.tsv` | - | - |
| `sequence_features.tsv` (1) | `ORF_TPMs_vs_features.tsv` | - | - |
| `features.pdf` | `ORF_TPMs_vs_features.pdf` | - | - |
| `codon_ribodens.tsv` | ? | - | - |
| `codon_ribodens.pdf` | ? | - | - |
| `startcodon_ribogridbar.pdf` | ? | - | - |
| `startcodon_ribogrid.pdf` | ? | - | - |
| `3ntframe_propbygene.pdf` |  ? | - | - |
| `3ntframe_bygene.tsv` | ? | - | - |
| `3ntframe_bygene_filtered.tsv` (2) | ? | - | - |
| `codon_ribodens_gathered.tsv` (2) | ? | - | - |
| `gene_position_length_counts_5start.tsv` (2) | ? | - | - |
| `read_lengths.pdf` | ? | - | - |
| `read_lengths.tsv` | ? | - | - |
| `minus.bedgraph` | ? | - | - |
| `plus.bedgraph` | ? | - | - |
| `pos_sp_nt_freq.tsv` | ? | - | - |
| `tpms.tsv` | ? | - | - |
| `<SAMPLE_ID>.bam` | ? | - | - |
| `<SAMPLE_ID>.bam.bai` | ? | - | - |
| `<SAMPLE_ID>.h5` | ? | - | - |
| `<SAMPLE_ID>_output_report.html` (3) | ? | - | - |

(1) Need to add these to `docs/user/prep-riboviz-operation.md` and output file listing in `docs/user/run-vignette.md`.
(2) Need to add these to `docs/user/prep-riboviz-operation.md`, `docs/user/riboviz_outputs.md` and output file listing in `docs/user/run-vignette.md`.
(3) Need to add these to output file listing in `docs/user/run-vignette.md`.

**Comment by mikej888** [comment]
@3mma-mack, @swinterbourne, here's a suggested process for updating each file name in turn, using the above table to track who is working on which file name.

Process for updates:

* Pick a file to rename. Edit the above comment and add your name to the 'Developer' column of the table.
* Update file names and file name patterns in code, which includes:
  - `riboviz/workflow_files.py` (file names and  file name patterns)
  - `riboviz/workflow_r.py` (file names and file name patterns)
  - Changing these should take care of the tests.
  - `rscripts/generate_stats_figs.R`
  - `rmarkdown/AnalysisOutputs.Rmd` (invoked by Nextflow `staticHTML` process)
  - `prep_riboviz.nf` (file names in `generateStatsFigs`, `renameTpms`, `collateTpms`, `countReads`, `staticHTML` processes),
  - `grep` for other occurrences.
* Run tests.
  - For the integration tests, tests associated with renamed files will fail as the integration test data directory (for example `test-data-20210804` from https://github.com/riboviz/test-data-20210804) will still have files with the old names.
  - You can get round this by, in your integration test data directory, creating a copy of the file which has the new name.
* Update Python and R constant and variable names and Nextflow variable and channel names to be consistent with the new file names (see earlier comment on cosmetic changes).
* Run tests.
* Update docs, which include:
  - `docs/user/riboviz_outputs.md`
  - `docs/user/prep-riboviz-operation.md` (add link to `docs/user/riboviz_outputs.md` above)
  - `docs/user/run-vignette.md` (listings of output files)
  - `docs/user/prep-riboviz-run-nextflow.md` (refers to `TPMs_collated.tsv` only)
  - `docs/user/run-dedup-demultiplex-examples.md` (refers to `codon_ribodens.pdf|tsv` only)
  - `docs/developer/debugging.md` (refers to `WTnone.h5` and `TPMs_collated.tsv` only)
  - `grep` for other occurrences.
* Commit.
* Pick another file name and repeat!

If you'd like a finer grained level of commits, then commit after the first test run, then again after the second, then again after updating the docs.

Changes to the following can be deferred:

* `docs/developer/testing.md`. This has sample output from an integration test run and can be recreated when the changes have been completed.
* `docs/images/*.dot`. Once all updates have been done I can update these and regenerate the associated `*.svg` image files.

@3mma-mack, @swinterbourne, feel free to ping me if you want me to review any commits you make.

**Comment by 3mma-mack** [comment]
Thanks @mikej888, that is all really clear and will make my job a lot easier

I'll let you know if I run into any problems

**Comment by kavousan** [comment]
@mikej888 this is a most useful recipe. Does it have a purpose outside this exercise? I just think it will be a shame to lose it in this ticket.

**Comment by mikej888** [comment]
@kavousan it's pretty specific to this task. However, there is a guide for adding configuration parameters, `docs/developer/config.md`, which would benefit from being expanded with some of the foregoing content (around testing, choice of variable, constant and channel names and where else to update docs).

**Comment by 3mma-mack** [comment]
I have updated the file and variable names for the files where new names were available in branch update-file-names-411. I ran tests as described by @mikej888 and all were as expected. The vignette still runs and produces the expected files, but with new names

I did not change the references to files in the .dot or .svg files, nor did I change the docs/developer/testing.md file

**Comment by 3mma-mack** [comment]
I have added a link between docs/user/prep-riboviz-operation.md and docs/user/riboviz_outputs.md in branch output-docs-395

**Comment by FlicAnderson** [comment]
@3mma-mack great - thanks for that, it'll definitely make life easier when we're testing this code out :)

**Comment by 3mma-mack** [comment]
I have merged develop into output-docs-395 and then output-docs-395 into update-file-names-411 so that there is a copy of riboviz_outputs.md on branch update-file-names-411.

I plan to update riboviz_outputs.md on branch update-file-names-411 so the names of files listed match the new file names where they have been changed

**Comment by 3mma-mack** [comment]
I have updated the names of the files that have been renamed so far in docs/user/[riboviz_outputs.md](https://github.com/riboviz/riboviz/blob/update-file-names-411/docs/user/riboviz_outputs.md) and I will change more as more name changes are suggested.

**Comment by 3mma-mack** [comment]
The names of files where new names had been suggested have all been updated in the code and documentation. 
Tests were all as expected and the vignette now runs to produce the following outputs:

```
(riboviz) [s1734289@node3a01(eddie) WT3AT]$ ls
frame_proportions_per_ORF.pdf               metagene_start_ribogrid_by_length.pdf           normalized_density_APEsites_per_codon.pdf  ORF_TPMs_vs_features.tsv         read_frame_per_ORF.tsv
gene_position_length_counts_5start.tsv      metagene_start_stop_read_counts.pdf             normalized_density_APEsites_per_codon.tsv  plus.bedgraph                    WT3AT.bam
metagene_normalized_profile_start_stop.pdf  metagene_start_stop_read_counts.tsv             nt_freq_per_read_position.tsv              read_counts_by_length.pdf        WT3AT.bam.bai
metagene_normalized_profile_start_stop.tsv  minus.bedgraph                                  ORF_TPMs_and_counts.tsv                    read_counts_by_length.tsv        WT3AT.h5
metagene_start_barplot_by_length.pdf        normalized_density_APEsites_per_codon_long.tsv  ORF_TPMs_vs_features.pdf                   read_frame_per_ORF_filtered.tsv  WT3AT_output_report.html
```

It may take a while to adjust to the new names as I am so used to the old ones, but they are definitely more descriptive.

**Comment by mikej888** [comment]
@3mma-mack, that's really good!

I've grepped for 'tsv' and 'pdf' in all of the following to look for any strays:

```
riboviz/workflow_files.py
riboviz/workflow_r.py
docs/**/*.md
riboviz/**.py
rscripts/**/*.R
rmarkdown/*
prep_riboviz.nf
```

I commited a couple of formatting tweaks - b9afcd3, d125d41 - and updated `TPMs_collated.tsv` to `TPMs_all_CDS_all_samples.tsv` in `collate_tpms.R` and `test_collate_tpms.R` - 6c5afd3.

For running the tests, to check, I created a custom test directory with the new file names as follows:

```console
cp -r test-data-20210804 test-data-20210804-411
cd test-data-20210804-411/output/
mv TPMs_collated.tsv TPMs_all_CDS_all_samples.tsv
mv read_counts.tsv read_counts_per_file.tsv
cd WTnone/
mv 3nt_periodicity.pdf metagene_start_stop_read_counts.pdf
mv 3nt_periodicity.tsv metagene_start_stop_read_counts.tsv
mv pos_sp_rpf_norm_reads.pdf metagene_normalized_profile_start_stop.pdf
mv pos_sp_rpf_norm_reads.tsv metagene_normalized_profile_start_stop.tsv
mv sequence_features.tsv ORF_TPMs_vs_features.tsv
mv features.pdf ORF_TPMs_vs_features.pdf
mv codon_ribodens.tsv normalized_density_APEsites_per_codon.tsv
mv codon_ribodens.pdf normalized_density_APEsites_per_codon.pdf
mv startcodon_ribogridbar.pdf metagene_start_barplot_by_length.pdf
mv startcodon_ribogrid.pdf metagene_start_ribogrid_by_length.pdf
mv 3ntframe_propbygene.pdf frame_proportions_per_ORF.pdf
mv 3ntframe_bygene.tsv read_frame_per_ORF.tsv
mv 3ntframe_bygene_filtered.tsv read_frame_per_ORF_filtered.tsv
mv codon_ribodens_gathered.tsv normalized_density_APEsites_per_codon_long.tsv
mv read_lengths.pdf read_counts_by_length.pdf
mv read_lengths.tsv read_counts_by_length.tsv
mv pos_sp_nt_freq.tsv nt_freq_per_read_position.tsv
mv tpms.tsv TPMs_all_CDS_all_samples.tsv ORF_TPMs_and_counts.tsv
cd ../WT3AT
# ...as for WTnone...
```
I then reran the integration tests then all the R and Python tests:
```console
$ rm -rf work vignette/index vignette/tmp vignette/output
$ nextflow run prep_riboviz.nf -ansi-log false -params-file vignette/vignette_config.yaml
$ pytest -vs riboviz/test/integration/test_integration.py --expected=../test-data-20210804-411 --skip-workflow
...
================= 50 passed, 46 skipped, 315 warnings in 1.32s =================
$ Rscript rscripts/tests/testthat.R
[ FAIL 0 | WARN 0 | SKIP 0 | PASS 522 ]
$ pytest --ignore-glob="*integration*"
...
================ 395 passed, 672 warnings in 3729.01s (1:02:09) ================
```
All fine :-)

One final batch of changes that could be done is to change the parameter names in `rmarkdown/AnalysisOutputs.Rmd`:

* `three_nucleotide_periodicity_data_file` => `metagene_start_stop_read_counts_data_file`
* `read_length_data_file` => `read_counts_by_length_data_file`
* `gene_read_frames_filtered_data_file` => `read_frame_per_orf_filtered_data_file`

Also the variables into which the contents of these files are read:

* `three_nucleotide_periodicity_data`
* `read_length_data`
* `gene_read_frame_data_filtered`

Then, make the parameter updates to `staticHTML` in `prep_riboviz.nf`:

```
process staticHTML {
    shell:
      ...
      script += "three_nucleotide_periodicity_data_file = '\$PWD/${sample_metagene_start_stop_read_counts_tsv}', "
      ...
      script += "read_length_data_file='\$PWD/${sample_read_counts_by_length_tsv}', "
      ...
      if (is_asite_disp_length_file) {
          script += ", gene_read_frames_filtered_data_file='\$PWD/${sample_read_frame_per_orf_filtered_tsv}'"
      }
      ...
}
```

I'll try and do the `doc/images/*.dot` and `*.svg` updates in the next couple of days.

**Comment by mikej888** [comment]
@3mma-mack wrote:

> There is currently an argument in the config file for `do_pos_sp_nt_freq`. I am not going to change this argument or references to it in the code at this point, despite it being relevant to the `pos_sp_nt_freq.tsv` file as it is an argument to determine whether to produce the file, and is not involved in the naming of the file.

We should discuss this at the next developer meeting and whether to update this to something like `output_metagene_normalized_profile`.

**Comment by mikej888** [comment]
Commits up to b0d7471 include updates to `docs/images/*.dot|.svg` with new output file names.

**Comment by 3mma-mack** [comment]
That is great, thank you for double checking everything and updating the `docs/images/*.dot|.svg` files.



> @3mma-mack wrote:
> 
> > There is currently an argument in the config file for `do_pos_sp_nt_freq`. I am not going to change this argument or references to it in the code at this point, despite it being relevant to the `pos_sp_nt_freq.tsv` file as it is an argument to determine whether to produce the file, and is not involved in the naming of the file.
> 
> We should discuss this at the next developer meeting and whether to update this to something like `output_metagene_normalized_profile`.

And I agree, this is something to discuss at the meeting

**Comment by 3mma-mack** [comment]
I have rerun the tests having pulled through the changes made by @mikej888 to Eddie. The tests still complete as expected which is good.

**Comment by mikej888** [comment]
We agreed at the developer meeting to rename  `do_pos_sp_nt_freq` to `output_metagene_normalized_profile`. Please see https://github.com/riboviz/riboviz/blob/develop/docs/developer/config.md

**Comment by FlicAnderson** [comment]
Deleting this branch as this issue is closed, the PR (#422) has been successfully merged into develop & I'm tidying up branches we don't need :)
